src/node_quic_stream.cc

```diff
@@ -379,6 +374,46 @@ BaseObjectPtr<QuicStream> QuicStream::New(
   return stream;
 }
 
+void QuicStream::BeginHeaders() {
+  Debug(this, "Beginning Headers");
+  // Upon start of a new block of headers, ensure that any
+  // previously collected ones are cleaned up.
+  headers_.clear();
+}
+
+bool QuicStream::AddHeader(std::unique_ptr<Header> header) {
+  Debug(this, "Header Added");
+  headers_.push_back(std::move(header));
+  // TODO(@jasnell): We need to limit the maximum number of headers.
+  // If the maximum count is reached, return false here instead.
+  return true;
+}
+
+void QuicStream::EndHeaders() {
+  Debug(this, "End Headers");
+  // Upon completion of a block of headers, convert the
+  // vector of Header objects into an array of name+value
+  // pairs, then call the on_stream_headers function.
+  std::vector<Local<Value>> headers;
+  for (auto&& header : headers_) {
+    // name and value should never be empty here, and if
+    // they are, there's an actual bug so go ahead and crash
+    Local<Value> pair[] = {
+      header->GetName(env()).ToLocalChecked(),
+      header->GetValue(env()).ToLocalChecked()
```

Comment:
If this is using `.ToLocalChecked()`, maybe the implementation in the `Header` base class should use `= 0` for `GetName()` and `GetValue()` rather than returning empty values that would lead to crashes here?